patch/go.proto

```diff
@@ -21,6 +21,12 @@ message Options {
 	// See https://golang.org/ref/spec#Struct_types.
 	optional bool embed = 2;
 
+	// The casstype option changes the generated field type.
+	// All generated code assumes that this type is castable to the protocol buffer field type,
+	// so it does not work for messages types.
+	// Repeated fields are also supported, both as '[]Type' or 'Types' where Types is a named slice type.
+	optional string casttype = 3;
```

Comment:
I'd prefer to call this `type` instead of `casttype`.

I know that GoGo called it that, but this isn't GoGo.